flink/src/test/java/org/apache/iceberg/flink/TestFlinkFilters.java

```diff
@@ -0,0 +1,306 @@
+      .field("field4", DataTypes.DOUBLE())
+      .field("field5", DataTypes.STRING())
+      .field("field6", DataTypes.BOOLEAN())
+      .field("field7", DataTypes.BINARY(10))
+      .field("field8", DataTypes.DECIMAL(10, 2))
+      .field("field9", DataTypes.DATE())
+      .field("field10", DataTypes.TIME())
+      .field("field11", DataTypes.TIMESTAMP())
+      .field("field12", DataTypes.TIMESTAMP_WITH_TIME_ZONE())
+      .field("field13", DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE())
+      .build();
+
+  @Test
+  public void testFlinkDataTypeEqual() {
+    matchLiteral("field1", 1, 1);
+    matchLiteral("field2", 10L, 10L);
+    matchLiteral("field3", 1.2F, 1.2F);
+    matchLiteral("field4", 3.4D, 3.4D);
+    matchLiteral("field5", "abcd", "abcd");
+    matchLiteral("field6", true, true);
+    matchLiteral("field7", new byte[] {'a', 'b'}, new byte[] {'a', 'b'});
+    matchLiteral("field8", BigDecimal.valueOf(10), BigDecimal.valueOf(10));
+
+    LocalDate date = LocalDate.parse("2020-12-23");
+    matchLiteral("field9", date, DateTimeUtil.daysFromDate(date));
+
+    LocalTime time = LocalTime.parse("12:13:14");
+    matchLiteral("field10", time, DateTimeUtil.microsFromTime(time));
+
+    LocalDateTime dateTime = LocalDateTime.parse("2020-12-23T12:13:14");
+    matchLiteral("field11", dateTime, DateTimeUtil.microsFromTimestamp(dateTime));
+
+    Instant instant = Instant.parse("2020-12-23T12:13:14.00Z");
+    matchLiteral("field12", instant, DateTimeUtil.microsFromInstant(instant));
+
+    matchLiteral("field13", instant, DateTimeUtil.microsFromInstant(instant));
+  }
+
+  @Test
+  public void testEquals() {
+    UnboundPredicate<Integer> expected = org.apache.iceberg.expressions.Expressions.equal("field1", 1);
+
+    org.apache.iceberg.expressions.Expression actual =
+        FlinkFilters.convert(resolve(Expressions.$("field1").isEqual(Expressions.lit(1)))).orElse(null);
+    assertPredicatesMatch(expected, (UnboundPredicate) actual);
+
+    org.apache.iceberg.expressions.Expression actual1 =
+        FlinkFilters.convert(resolve(Expressions.lit(1).isEqual(Expressions.$("field1")))).orElse(null);
+    assertPredicatesMatch(expected, (UnboundPredicate) actual1);
+  }
+
+  @Test
+  public void testEqualsNaN() {
+    UnboundPredicate<Float> expected = org.apache.iceberg.expressions.Expressions.isNaN("field3");
+
+    org.apache.iceberg.expressions.Expression actual =
+        FlinkFilters.convert(resolve(Expressions.$("field3").isEqual(Expressions.lit(Float.NaN)))).orElse(null);
+    assertPredicatesMatch(expected, (UnboundPredicate) actual);
+
+    org.apache.iceberg.expressions.Expression actual1 =
+        FlinkFilters.convert(resolve(Expressions.lit(Float.NaN).isEqual(Expressions.$("field3")))).orElse(null);
+    assertPredicatesMatch(expected, (UnboundPredicate) actual1);
+  }
+
+  @Test
+  public void testNotEquals() {
+    UnboundPredicate<Integer> expected = org.apache.iceberg.expressions.Expressions.notEqual("field1", 1);
```

Comment:
It's good to have test cases for all other data types except `Integer`.  I guess we may need few abstraction so that the test code won't duplicate too many.

Reply:
I added all type test for `testNotEquals ` and `testEquals `